app/models/spree_amazon/order.rb

```diff
@@ -25,6 +27,24 @@ def confirm
     mws.confirm_order
   end
 
+  def close_order_reference!
+    response = mws.close_order_reference
+    parsed_response = Hash.from_xml(response.body) rescue nil
+
+    if response.success
+      true
+    else
+      parsed_response = Hash.from_xml(response.body) rescue nil
+      message = if parsed_response && parsed_response['ErrorResponse']
+        error = parsed_response.fetch('ErrorResponse').fetch('Error')
+        "#{response.code} #{error.fetch('Code')}: #{error.fetch('Message')}"
```

Comment:
@manmartinez That was intentional -- If an `ErrorResponse` key exists then afaik it should always have have an `Error.Code` and an `Error.Message`.  If it doesn't then I think I'd like for something to blow up so I find out about it.  Another reason I think #32 would be great, as you say.  This in particular seems like it should live in the amazon gem (easy & robust access to error messages & codes).